cetmix_tower_server/models/cx_tower_reference_mixin.py

```diff
@@ -126,15 +147,42 @@ def create(self, vals_list):
         Returns:
             Records: The created record(s).
         """
-        if not self._context.get("reference_mixin_override"):
+
+        if vals_list and not self._context.get("reference_mixin_override"):
+            # Check if we need to populate references based on parent record
+            auto_generate_settings = self._get_pre_populated_model_data().get(
+                self._name
+            )
+            if auto_generate_settings:
+                parent_model, relation_field, suffix = auto_generate_settings
+                vals_list = self._pre_populate_references(
+                    parent_model, relation_field, vals_list, suffix
+                )
+
+            # Fix or create references
             for vals in vals_list:
-                if vals.get("name"):
-                    vals["name"] = vals["name"].strip()
+                if not vals:
+                    continue
+
+                # Remove leading and trailing whitespaces from name
+                vals_name = vals.get("name", "")
+                name = vals_name.strip()
+
+                # Remove leading and trailing whitespaces from reference
+                reference = vals.get("reference", "").strip()
+
```

Comment:
We need both variables because they are compared in line 179
```python
                if vals_name != name:
                    vals["name"] = name
```